app/src/features/FhirResourceTree/TreeItem.tsx

```diff
@@ -1,19 +1,98 @@
 import React, { useState } from "react";
 
-import { Icon } from "@blueprintjs/core";
+import { Icon, MaybeElement } from "@blueprintjs/core";
 import { IconName, IconNames } from "@blueprintjs/icons";
-import { Typography } from "@material-ui/core";
+import {
+  IconButton as MuiIconButton,
+  makeStyles,
+  Theme,
+  Typography,
+} from "@material-ui/core";
 import { TreeItem as MuiTreeItem } from "@material-ui/lab";
+import clsx from "clsx";
 
 import useFhirResourceTreeData from "common/hooks/useFhirResourceTreeData";
 
 import { ElementNode } from "./resourceTreeSlice";
 
 type TreeItemProps = {
   elementNode: ElementNode;
+  isArrayItem?: boolean;
 };
 
-const TreeItem = ({ elementNode }: TreeItemProps): JSX.Element => {
+const useStyle = makeStyles((theme: Theme) => ({
+  root: {
+    "& .MuiTreeItem-content .MuiTreeItem-label": {
+      borderRadius: 4,
+      height: 32,
+      paddingLeft: 10,
+    },
+  },
+  treeItemContainer: {
+    display: "flex",
+    alignItems: "center",
+    height: "100%",
+    "&:hover": {
+      "& button": {
+        display: "flex",
+      },
+    },
+  },
+  icon: {
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    fill: theme.palette.text.primary,
+  },
+  treeItemTitle: {
+    fontWeight: 500,
+  },
+  treeItemType: {
+    fontWeight: 400,
+    flexGrow: 1,
+    overflow: "hidden",
+    whiteSpace: "nowrap",
+    width: 1,
+    textOverflow: "ellipsis",
+    display: "inline-block",
+  },
+  margin: {
+    marginLeft: theme.spacing(1),
+  },
+  iconRight: {
+    fill: theme.palette.text.secondary,
+  },
+  labelEndIcons: {
+    display: "none",
+    marginRight: 4,
+  },
+}));
+
+type IconButtonProps = {
+  icon: IconName | MaybeElement;
+  onClick: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+};
+
+const IconButton = ({ icon, onClick }: IconButtonProps): JSX.Element => {
+  const classes = useStyle();
+  return (
+    <MuiIconButton
+      size="small"
+      className={classes.labelEndIcons}
+      onClick={onClick}
+    >
+      <Icon
+        className={clsx(classes.icon, classes.iconRight)}
+        icon={icon}
+        iconSize={15}
+      />
+    </MuiIconButton>
+  );
+};
```

Comment:
I'd prefer if you made a new file for this in `common/components/IconButton.tsx`